src/loaders/express/tests/helmet.spec.ts

```diff
@@ -0,0 +1,169 @@
+    mockHelmet.ieNoOpen = jest.fn().mockImplementation(() => 'ieNoOpen')
+    mockHelmet.dnsPrefetchControl = jest
+      .fn()
+      .mockImplementation(() => 'dnsPrefetchControl')
+    mockHelmet.hidePoweredBy = jest
+      .fn()
+      .mockImplementation(() => 'hidePoweredBy')
+    mockHelmet.referrerPolicy = jest
+      .fn()
+      .mockImplementation(() => 'referrerPolicy')
+    mockHelmet.contentSecurityPolicy = jest
+      .fn()
+      .mockImplementation(() => 'contentSecurityPolicy')
+    helmetMiddlewares()
+    expect(mockHelmet.xssFilter).toHaveBeenCalled()
+    expect(mockHelmet.noSniff).toHaveBeenCalled()
+    expect(mockHelmet.ieNoOpen).toHaveBeenCalled()
+    expect(mockHelmet.dnsPrefetchControl).toHaveBeenCalled()
+    expect(mockHelmet.hidePoweredBy).toHaveBeenCalled()
+    expect(mockHelmet.referrerPolicy).toHaveBeenCalled()
+    expect(mockHelmet.contentSecurityPolicy).toHaveBeenCalled()
+  })
+
+  it('should call helmet.contentSecurityPolicy() with the correct cspCoreDirectives', () => {
+    const cspCoreDirectives = {
+      defaultSrc: ["'self'"],
+      imgSrc: [
+        "'self'",
+        'data:',
+        'https://www.googletagmanager.com/',
+        'https://www.google-analytics.com/',
+        `https://s3-${config.aws.region}.amazonaws.com/agency.form.sg/`,
+        config.aws.imageBucketUrl,
+        config.aws.logoBucketUrl,
+        '*',
+      ],
+      fontSrc: ["'self'", 'data:', 'https://fonts.gstatic.com/'],
+      scriptSrc: [
+        "'self'",
+        'https://www.googletagmanager.com/',
+        'https://ssl.google-analytics.com/',
+        'https://www.google-analytics.com/',
+        'https://www.tagmanager.google.com/',
+        'https://www.google.com/recaptcha/',
+        'https://www.recaptcha.net/recaptcha/',
+        'https://www.gstatic.com/recaptcha/',
+        'https://www.gstatic.cn/',
+        'https://www.google-analytics.com/',
+      ],
+      connectSrc: [
+        "'self'",
+        'https://www.google-analytics.com/',
+        'https://ssl.google-analytics.com/',
+        'https://sentry.io/api/',
+        config.aws.attachmentBucketUrl,
+        config.aws.imageBucketUrl,
+        config.aws.logoBucketUrl,
+      ],
+      frameSrc: [
+        "'self'",
+        'https://www.google.com/recaptcha/',
+        'https://www.recaptcha.net/recaptcha/',
+      ],
+      objectSrc: ["'none'"],
+      styleSrc: [
+        "'self'",
+        'https://www.google.com/recaptcha/',
+        'https://www.recaptcha.net/recaptcha/',
+        'https://www.gstatic.com/recaptcha/',
+        'https://www.gstatic.cn/',
+        // For inline styles in angular-sanitize.js
+        "'sha256-b3IrgBVvuKx/Q3tmAi79fnf6AFClibrz/0S5x1ghdGU='",
+      ],
+      formAction: ["'self'"],
+    }
+    helmetMiddlewares()
+    expect(mockHelmet.contentSecurityPolicy).toHaveBeenCalledWith(
+      expect.objectContaining({
+        directives: expect.objectContaining({
+          ...cspCoreDirectives,
+        }),
```

Comment:
I'm wondering
(1) if we need to make this test stricter. shouldn't we expect the function to be called with exactly the object we want? might be better to declare the base object `cspCoreDirectives`, add `cspReportUri` and `upgradeInsecureRequests` as relevant, then expect the function to have been called with the exact object we expect.
(2) if we can make this test syntax less convoluted. perhaps (1) will solve that by itself

Reply:
I've simplified the tests and made them stricter